### DISCUSSION TITLE
Feature Request

### DISCUSSION THREAD

**Discussion by stav242** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#389] [Ideas]
I have the existing project installed and working fine. *thanks*

I am now installing a second esp32-wroom with a Robodyn 16A AC dimmer control - so i can regulate the heater power.

This will varied by Homeassistant / nodered to utilise my excess solar power to heat up the hot-tub water.

My request is.
can you integrate the robodyn code into the existing project so i can use just this single project board to control the robodyn ac dimmer as well ?

- it just needs 2 pins and the code and then expose it via MQTT to ha.

**Comment by visualapproach** [comment]
You can try. D0 and D8 are exposed. Only D8 can do PWM.

**Comment by stav242** [comment]
Ok I was hoping I could encourage someone to help me... Im a bit of a
novice with any programming. I can manage to fumble through other people's
scripts to get things working but I don't think I could adapt your code to
make this work. It's a bit too far for my skills. Im happy with electronics
side.

Any chance you could help ?

**Reply by visualapproach** [reply]
Sorry, I don't really have the time to do that. Here is some ideas of where to begin:
`#include ` that library for your device into main.cpp.
look at the examples in that library and import what you need into this code. There is a solar-example in there.
Make a web server handler, or mqtt handler unless you want to add commands to my library. I.e. react to "layzspa.local/heaterON" calls.

I'm not sure it will play nicely with the rest of the code. It uses interrupts which may interfere with the communication to the spa. But you never know until you try.
Wish you luck

**Comment by stav242** [comment]
ok, ie got it working on a separate ESP32 atm, ill probably leave it as is
for the time being until i decide to delve into it further.
thanks.
